api/base_controller.go

```diff
@@ -192,23 +193,17 @@ func (c *BaseController) CreateObject(r *web.Request) (*web.Response, error) {
 		ResourceType:  c.objectType,
 		PlatformID:    types.SMPlatform,
 		CorrelationID: log.CorrelationIDFromContext(ctx),
-		Context:       &types.OperationContext{Async: c.shouldExecuteAsync(r)},
+		Context:       c.getOperationContextByRequest(r),
 	}
 
-	if c.shouldExecuteAsync(r) {
-		log.C(ctx).Debugf("Request will be executed asynchronously")
+	if operation.IsAsyncResponse() {
```

Comment:
in this case you still don't have the response from the broker
it means that e.Context.BrokerResponse.Async will be always false
so you can just check the operation.Context.Async flag

Reply:
This is needed 2 reflect a case where the client did not send the **async=true/false** 
Broker's actual response is checked later on and is stored in the BrokerResponse.async context-param.